### PR TITLE
Automatically increment version and add changelog 

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

File: eng/scripts/Update-CliVersion.ps1 [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+if (!$version) {
+    # Increment after release
+    $version = getVersion
+
+    if ($version.IsPrerelease) {
+        if ($version.HasValidPrereleaseLabel()) {
+            # 0.1.0-beta.1 -> 0.1.0-beta.2
+            $version.PrereleaseNumber++
+        } else {
+            # 0.1.0 -> 0.2.0-beta.1
```
Comment (ellismg) [inline eng/scripts/Update-CliVersion.ps1:28]: Is this example correct?  I'm trying to understand how `0.1.0` would ever be considered pre-release and we'd go down this path.  I think maybe the example should be `0.1.0-badPrereleaseLabel.1 -> 0.2.0-beta.1`?

File: eng/scripts/Update-CliVersion.ps1 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            $version.Minor++
+            $version.PrereleaseLabel = 'beta'
+            $version.PrereleaseNumber = 1
+        }
+    } else {
+        # 1.0.0-beta.1 -> 1.0.0-beta.2
```
Comment (ellismg) [inline eng/scripts/Update-CliVersion.ps1:34]: Feels like this example is wrong, since 1.0.0-beta.1 would be a pre-release version right?

Comment (danieljurek) [reply eng/scripts/Update-CliVersion.ps1:34]: Indeed this scenario is properly covered on line 24

File: eng/scripts/Update-CliVersion.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/Update-CliVersion.ps1:28]: From: https://semver.org/#spec-item-4 

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

The semver script that we have today lumps "initial development" in as a subset of "prerelease" though a good argument could probably be made that "initial development" is not pre-release. If we need to treat the two differently, we can do that. 

This does also handle the scenario you described.

**Comment by danieljurek** [comment]
Closing in favor of https://github.com/Azure/azure-dev/pull/814

**Pull request closed without merging**